### PR TITLE
Module 13

### PR DESCRIPTION
**Pull request by JoeFisherW**

### CODE REVIEW

**Comment by JoeFisherW** [comment]
image

File: main.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ terraform {
     resource_group_name = "SoftwirePilot_JosephFisher_ProjectExercise"
     storage_account_name = "storageaccountjosfis"
     container_name = "todoapp-container"
-    key = "8O1ytbiAe65qZQZS3Sx9pDppNbJWYWSTFfklEwcVN3p0ZFbxLn4YZ5I5vv9sejmqFdrFrTDwr39xhJMabdG87Q=="
+    key = "TodoAppAccessKey"
```
Comment (BenRamchandani) [inline main.tf:6]: You should rotate this key, it's still valid.

File: main.tf [hunk 2]
```diff
@@ -74,7 +74,8 @@ resource "azurerm_app_service" "main" {
   "FLASK_ENV" = "developement"
   "LOAD_DISABLED" = ""
   "SECRET_KEY" = "real_key"
-  "Mongo_db" = "todo_db"
+  "LOG_LEVEL" = "DEBUG"
+  "Mongo_db" = "todoapp-cosmos-mongo-db"
```
Comment (BenRamchandani) [inline main.tf:78]: Would be nice to have Loggly set up for the deployed app as well, but this isn't required.

File: app.py [hunk 3]
```diff
@@ -38,11 +47,13 @@ def index():
     @login_required
     def add_todo():
         if current_user.is_active == True:
-            if current_user.role == 'writer' or current_user == 'admin':
+            if current_user.role == 'writer' or current_user.role == 'admin':
                 mongo.create_new_item(request.form.get('title'))
+                app.logger.info("User %s added Todo item '%s'", current_user.name, request.form.get('title'))
```
Comment (BenRamchandani) [inline app.py:52]: Would be nice to log the created item's id too, since that's what you log elsewhere (definitely optional)

File: app.py [hunk 4]
```diff
@@ -38,11 +47,13 @@ def index():
...
                 return redirect('/')
             else :
                 flash('You do not have access. Please contact an admin')
+                app.logger.info("User %s attempted to add Todo item, incorrect permissions", current_user.name)
```
Comment (BenRamchandani) [inline app.py:56]: I think `warn` or maybe even `error` would be appropriate here.

File: .env.template [hunk 5]
```diff
@@ -3,4 +3,5 @@ Mongo_db = example_db
 PORT=5000
 clientId=GitHubClientId
 client_secret_m10=GitHubOAuthAppClientSecret
+LOGGLY_TOKEN
```
Comment (BenRamchandani) [inline .env.template:6]:
```suggestion
client_secret=GitHubOAuthAppClientSecret
LOGGLY_TOKEN=
LOG_LEVEL=DEBUG
```
Doesn't run without a log level

**Review by BenRamchandani — COMMENTED** [review]
It would be good to set up Loggly for the terraformed app service, and include some details about how to set it up in your README.

The logging you've added looks sensible though, so happy to accept, the comments below are optional.

**Pull request merged**